### PR TITLE
Make Impositio Manus always overwrites itself

### PR DESCRIPTION
**Pull request by Kenpachi2k13**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Impositio Manus always overwrites itself regardless of which skill level is used.
_The [skill rebalancing update](https://ro.gnjoy.com/news/notice/View.asp?seq=7040) changed this behaviour, but yet we didn't even start implementing that update._

**Issues addressed:** #695

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

File: src/map/status.c [hunk 1]
```diff
@@ -10507,273 +10507,273 @@ static void status_change_start_stop_action(struct block_list *bl, enum sc_type
...
+		if (type != SC_SIREN) status_change_end(bl, SC_SIREN, INVALID_TIMER);
+		if (type != SC_DEEP_SLEEP) status_change_end(bl, SC_DEEP_SLEEP, INVALID_TIMER);
+		if (type != SC_SIRCLEOFNATURE) status_change_end(bl, SC_SIRCLEOFNATURE, INVALID_TIMER);
+		if (type != SC_LERADS_DEW) status_change_end(bl, SC_LERADS_DEW, INVALID_TIMER);
+		if (type != SC_MELODYOFSINK) status_change_end(bl, SC_MELODYOFSINK, INVALID_TIMER);
+		if (type != SC_BEYOND_OF_WARCRY) status_change_end(bl, SC_BEYOND_OF_WARCRY, INVALID_TIMER);
+		if (type != SC_UNLIMITED_HUMMING_VOICE) status_change_end(bl, SC_UNLIMITED_HUMMING_VOICE, INVALID_TIMER);
+		if (type != SC_GLOOMYDAY) status_change_end(bl, SC_GLOOMYDAY, INVALID_TIMER);
+		if (type != SC_SONG_OF_MANA) status_change_end(bl, SC_SONG_OF_MANA, INVALID_TIMER);
+		if (type != SC_DANCE_WITH_WUG) status_change_end(bl, SC_DANCE_WITH_WUG, INVALID_TIMER);
```
Comment (AnnieRuru) [inline src/map/status.c:10741]: if really apply hercules coding style isn't these lines should be
```c
		if (type != SC_SIREN)
			status_change_end(bl, SC_SIREN, INVALID_TIMER);
		if (type != SC_DEEP_SLEEP)
			status_change_end(bl, SC_DEEP_SLEEP, INVALID_TIMER);
```

see below, you done this correctly
```c
	case SC_GENTLETOUCH_ENERGYGAIN:
	case SC_GENTLETOUCH_CHANGE:
	case SC_GENTLETOUCH_REVITALIZE:
		if (type != SC_GENTLETOUCH_REVITALIZE)
			status_change_end(bl, SC_GENTLETOUCH_REVITALIZE, INVALID_TIMER);
		if (type != SC_GENTLETOUCH_ENERGYGAIN)
			status_change_end(bl, SC_GENTLETOUCH_ENERGYGAIN, INVALID_TIMER);
		if (type != SC_GENTLETOUCH_CHANGE)
			status_change_end(bl, SC_GENTLETOUCH_CHANGE, INVALID_TIMER);
		break;
```

**Review by AnnieRuru — CHANGES_REQUESTED** [review]

**Review by 4144 — APPROVED** [review]

File: src/map/status.c (hunk 1 above)
Comment (Kenpachi2k13) [reply src/map/status.c:10741]: Yay... I overlooked those. Thanks. 👍

Comment (Kenpachi2k13) [reply src/map/status.c]: Fixed.

File: src/map/status.c [hunk 2]
```diff
@@ -10507,273 +10507,319 @@ static void status_change_start_stop_action(struct block_list *bl, enum sc_type
...
+		if (type != SC_GENTLETOUCH_REVITALIZE)
+			status_change_end(bl, SC_GENTLETOUCH_REVITALIZE, INVALID_TIMER);
+
+		if (type != SC_GENTLETOUCH_ENERGYGAIN)
+			status_change_end(bl, SC_GENTLETOUCH_ENERGYGAIN, INVALID_TIMER);
+
+		if (type != SC_GENTLETOUCH_CHANGE)
+			status_change_end(bl, SC_GENTLETOUCH_CHANGE, INVALID_TIMER);
+
+		break;
```
Comment (AnnieRuru) [inline src/map/status.c:10812]: why there are so many empty lines ?
I guess its still editing

**Review by AnnieRuru — CHANGES_REQUESTED** [review]

File: src/map/status.c (hunk 2 above)
Comment (Kenpachi2k13) [reply src/map/status.c:10812]: Nope, that's final. Those blank lines improve the readability. (At least in my opinion.)

Comment (AnnieRuru) [reply src/map/status.c:10812]: I recheck the hercules coding style and ...
it only says leave empty lines for comments ... and each every new function

these line break ... doesn't fall into either of these ...
well if the code doesn't break then I trust you =/

**Review by AnnieRuru — APPROVED** [review]

File: src/map/status.c (hunk 2 above)
Comment (Kenpachi2k13) [reply src/map/status.c:10812]: The code wont break. It's just a clear visual separation between the `if`-clause and leaving the `switch`-case.

**Review by MishimaHaruna — APPROVED** [review]

**Pull request merged**